src/antsibull_core/args.py

```diff
@@ -16,27 +16,42 @@ class InvalidArgumentError(Exception):
     """A problem parsing or validating a command line argument."""
 
 
-def get_toplevel_parser(package, **kwargs) -> argparse.ArgumentParser:
+def get_toplevel_parser(
+    package: str,
+    *,
+    package_version: str | None = None,
+    program_name: str | None = None,
+    **kwargs,
+) -> argparse.ArgumentParser:
     """
     Create a toplevel argument parser with options common across all scripts.
 
     :arg package: The Python package containing this CLI program.
+    :arg package_version: If provided, use this instead of the version for ``package``
+        provided by ``importlib.metadata.version()``.
+    :arg program_name: If provided, show a more concrete description for this program.
     :args kwargs: This function takes any keyword arguments and passes them directly on to
         the :class:`argparse.ArgumentParser` constructor.
     :returns: :class:`argparse.ArgumentParser` with common script arguments added.
     """
-    try:
-        version = metadata.version(package)
-    except metadata.PackageNotFoundError:
-        # If there's no metadata foun, assume we're running from source
-        version = "source"
+    if package_version is None:
+        try:
+            package_version = metadata.version(package)
+        except metadata.PackageNotFoundError:
+            # If there's no metadata foun, assume we're running from source
+            package_version = "source"
+
+    if program_name is None:
+        program_name = package
 
     toplevel_parser = argparse.ArgumentParser(**kwargs)
     toplevel_parser.add_argument(
         "--version",
         action="version",
-        version=version,
-        help="Print the antsibull version",
+        version=package_version,
+        help=f"Print the {program_name} version"
+        if program_name
+        else "Print the program's version",
```

Comment:
Hmm, this `else` will only be reached if program_name is an empty string.

Reply:
It will also be reached if it is `None`, which is it's default value.

Reply:
You have

``` python
    if program_name is None:
        program_name = package
```

a couple lines above.

Reply:
Oops, I had that in for an earlier version and wanted to remove it. It's gone now.